### PR TITLE
New Example: Enforce AWS MFA via IAM

### PR DESCRIPTION
**Pull request by toddmichael**

### CODE REVIEW

File: aws/aws_iam_mfa.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
+locals {
+  aws_account = "account_number_goes_here"
+}
+
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
data "aws_caller_identity" "current" {}
```

File: aws/aws_iam_mfa.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  "Statement": [
+    {
+      "Sid": "AllowAllUsersToListAccounts",
+      "Effect": "Allow",
+      "Action": [
+        "iam:ListAccountAliases",
+        "iam:ListUsers"
+      ],
+      "Resource": [
+        "arn:aws:iam::${local.aws_account}:user/*"
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/*"
```

File: aws/aws_iam_mfa.tf [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        "iam:ChangePassword",
+        "iam:CreateLoginProfile",
+        "iam:DeleteLoginProfile",
+        "iam:GetAccountPasswordPolicy",
+        "iam:GetAccountSummary",
+        "iam:GetLoginProfile",
+        "iam:UpdateLoginProfile"
+      ],
+      "Resource": [
+        "arn:aws:iam::${local.aws_account}:user/$${aws:username}"
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
```

File: aws/aws_iam_mfa.tf [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    },
+    {
+      "Sid": "AllowIndividualUserToListTheirMFA",
+      "Effect": "Allow",
+      "Action": [
+        "iam:ListVirtualMFADevices",
+        "iam:ListMFADevices"
+      ],
+      "Resource": [
+        "arn:aws:iam::${local.aws_account}:mfa/*",
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/*",
```

File: aws/aws_iam_mfa.tf [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        "arn:aws:iam::${local.aws_account}:user/$${aws:username}"
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
```

File: aws/aws_iam_mfa.tf [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+      "Effect": "Allow",
+      "Action": [
+        "iam:CreateVirtualMFADevice",
+        "iam:DeactivateMFADevice",
+        "iam:DeleteVirtualMFADevice",
+        "iam:EnableMFADevice",
+        "iam:ResyncMFADevice"
+      ],
+      "Resource": [
+        "arn:aws:iam::${local.aws_account}:mfa/$${aws:username}",
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/$${aws:username}",
```

File: aws/aws_iam_mfa.tf [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        "arn:aws:iam::${local.aws_account}:user/$${aws:username}"
```
Comment (netflash) [inline aws/aws_iam_mfa.tf]:
```suggestion
        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
```

File: aws/aws_iam_mfa.tf (hunk 1 above)
Comment (toddmichael) [reply aws/aws_iam_mfa.tf]: Good call.  Updated.  To reduce keystrokes I kept `locals` and just set `aws_account` to the data resource.

**Pull request merged**